src/client/common/amlContext.ts

```diff
@@ -5,32 +5,20 @@ import { inject, injectable, named } from 'inversify';
 import { Memento } from 'vscode';
 import { IExtensionSingleActivationService } from '../activation/types';
 import { setSharedProperty } from '../telemetry';
-import { IWorkspaceService } from './application/types';
 import { GLOBAL_MEMENTO, IMemento } from './types';
 import { noop } from './utils/misc';
 
 const amlComputeMementoKey = 'JVSC_IS_AML_COMPUTE_INSTANCE';
 
 @injectable()
 export class AmlComputeContext implements IExtensionSingleActivationService {
-    constructor(
-        @inject(IMemento) @named(GLOBAL_MEMENTO) private readonly memento: Memento,
-        @inject(IWorkspaceService) private readonly workspace: IWorkspaceService
-    ) {}
+    constructor(@inject(IMemento) @named(GLOBAL_MEMENTO) private readonly memento: Memento) {}
 
     public get isAmlCompute() {
         return this.memento.get<boolean>(amlComputeMementoKey, false) || this.isAmlComputeWorkspace();
     }
     private isAmlComputeWorkspace() {
-        let isRemoteConnection: boolean = false;
-        // If there is at least 1 folder, we can check if the scheme and authority match a remote connection
-        if (Array.isArray(this.workspace.workspaceFolders) && this.workspace.workspaceFolders.length > 0) {
-            // We are in a remote connection to a CI if the scheme is vscode-remote and the authority contains amlext
-            isRemoteConnection = this.workspace.workspaceFolders.some(
-                (item) => item.uri.scheme === 'vscode-remote' && item.uri.authority.indexOf('amlext') >= 0
-            );
-        }
-        return isRemoteConnection;
+        return typeof process.env.AZURE_EXTENSION_DIR === 'string';
```

Comment:
Did the AML team point this out? I guess this works as far as I know, but it seems much broader versus the other check. This seems like it would always be set for all of VS Code if you AML installed?

Reply:
Nope, their solution didn't work. I'm using this for now.
Worst case scenario users on a VM with this variable will end up with Native Notebook UI instead of webview UI.

Hence done see much of an issue at this state, i.e. this is a hack.
FYI - @greazer & AML team have been informed of this, Worst case we revert this, at least this gets it working (again no harm).

Reply:
Found a better fix @IanMatthewHuff